api/src/legacy_soap_api/legacy_soap_api_auth.py

```diff
@@ -34,17 +34,26 @@ class SOAPClientCertificate(BaseModel):
     serial_number: int
     fingerprint: str
 
-    def get_pem(self, key_map: dict) -> tuple[str, str]:
+    def get_pem(self, key_map: dict, db_session: db.Session) -> str:
         """Note that this auth mechanism will only be configured in lower environments
 
         There will be no prod configurations for this auth mechanism.
         TODO - is the above true? I think this happens for prod as well?
         """
+
+        legacy_certificate_id = db_session.execute(
+            select(LegacyCertificate.legacy_certificate_id).where(
+                LegacyCertificate.serial_number == str(self.serial_number)
+            )
```

Comment:
We fetch the certificate elsewhere already, we shouldn't fetch it twice, is there a way we can fetch it once and pass it around? Perhaps attach it to this SOAPClientCertificate class? Might be tricky to change to work that way from the look of things because the place where we create the SoapClientCertificate is pretty nested in non-db code, but would simplify quite a bit if we could.